File: en/codes/csharp/chapter_hashing/hash_map.cs

/**
 * File: hash_map.cs
 * Created Time: 2022-12-23
 * Author: haptear (haptear@hotmail.com)
 */

namespace hello_algo.chapter_hashing;

public class hash_map {
    [Test]
    public void Test() {
        /* Initialize hash table */
        Dictionary<int, string> map = new() {
            /* Add operation */
            // Add key-value pair (key, value) to the hash table
            { 12836, "Ha" },
            { 15937, "Luo" },
            { 16750, "Suan" },
            { 13276, "Fa" },
            { 10583, "Ya" }
        };
        Console.WriteLine("\nAfter adding, the hash table is\nKey -> Value");
        PrintUtil.PrintHashMap(map);

        /* Query operation */
        // Enter key to the hash table, get value
        string name = map[15937];
        Console.WriteLine("\nEnter student ID 15937, found name " + name);

        /* Remove operation */
        // Remove key-value pair (key, value) from the hash table
        map.Remove(10583);
        Console.WriteLine("\nAfter removing 10583, the hash table is\nKey -> Value");
        PrintUtil.PrintHashMap(map);

        /* Traverse hash table */
        Console.WriteLine("\nTraverse key-value pairs Key->Value");
        foreach (var kv in map) {
            Console.WriteLine(kv.Key + " -> " + kv.Value);
        }
        Console.WriteLine("\nIndividually traverse keys Key");
        foreach (int key in map.Keys) {
            Console.WriteLine(key);
        }
        Console.WriteLine("\nIndividually traverse values Value");
        foreach (string val in map.Values) {
            Console.WriteLine(val);
        }
    }
}

File: en/codes/java/chapter_hashing/hash_map.java
/**
 * File: hash_map.java
 * Created Time: 2022-12-04
 * Author: krahets (krahets@163.com)
 */

package chapter_hashing;

import java.util.*;
import utils.*;

public class hash_map {
    public static void main(String[] args) {
        /* Initialize hash table */
        Map<Integer, String> map = new HashMap<>();

        /* Add operation */
        // Add key-value pair (key, value) to the hash table
        map.put(12836, "Ha");
        map.put(15937, "Luo");
        map.put(16750, "Suan");
        map.put(13276, "Fa");
        map.put(10583, "Ya");
        System.out.println("\nAfter adding, the hash table is\nKey -> Value");
        PrintUtil.printHashMap(map);

        /* Query operation */
        // Enter key to the hash table, get value
        String name = map.get(15937);
        System.out.println("\nEnter student ID 15937, found name " + name);

        /* Remove operation */
        // Remove key-value pair (key, value) from the hash table
        map.remove(10583);
        System.out.println("\nAfter removing 10583, the hash table is\nKey -> Value");
        PrintUtil.printHashMap(map);

        /* Traverse hash table */
        System.out.println("\nTraverse key-value pairs Key->Value");
        for (Map.Entry<Integer, String> kv : map.entrySet()) {
            System.out.println(kv.getKey() + " -> " + kv.getValue());
        }
        System.out.println("\nIndividually traverse keys Key");
        for (int key : map.keySet()) {
            System.out.println(key);
        }
        System.out.println("\nIndividually traverse values Value");
        for (String val : map.values()) {
            System.out.println(val);
        }
    }
}
